pkg/flow/flow.go

```diff
@@ -192,6 +202,7 @@ func newController(o controllerOptions) *Flow {
 					DataPath:          o.DataPath,
 					ID:                id,
 					ServiceMap:        serviceMap.FilterByName(availableServices),
+					WorkerPool:        workerPool,
```

Comment:
Is it intentional that modules share a worker pool with the root module? We'll have to see if this causes contention down the line as the number of total components grow across many, many modules.

Reply:
Yes, this was intentional. Having a worker pool per module introduces a problem of adequate pool sizing - e.g. one module may have 100s of components, while another just 1. It can also lead to too many goroutines if we have a lot of small modules. 

If we can use a single pool, and we run into the problem with too many modules, it may still be preferred to just expose a config for shared pool's thread count and queue sizes.